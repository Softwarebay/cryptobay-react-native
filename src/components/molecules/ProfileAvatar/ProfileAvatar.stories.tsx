import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';

import { ProfileAvatar } from './ProfileAvatar';

storiesOf('Molecules/ProfileAvatar', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Profile Avatar', () => (
    <ProfileAvatar onPressAvatar={action('onPressAvatar')} />
  ));
