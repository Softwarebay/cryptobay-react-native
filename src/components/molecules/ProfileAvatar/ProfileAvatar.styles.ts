import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  profileAvatar: {
    position: 'relative',
  },
});
