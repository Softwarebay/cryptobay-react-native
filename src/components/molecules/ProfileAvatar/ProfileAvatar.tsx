import React from 'react';
import { ImageSourcePropType, StyleProp, View, ViewStyle } from 'react-native';

import { Picture } from '../../atoms';
import { Icon } from '../../atoms/Icon/Icon';
import { Icons } from '../../atoms/Icon/icons.enum';

import { styles } from './ProfileAvatar.styles';

type ProfileAvatarProps = {
  pictureFile: ImageSourcePropType;
  pictureTestID?: string;
  iconTestID?: string;
};

export const ProfileAvatar: React.FC<ProfileAvatarProps> = ({
  pictureFile,
  pictureTestID,
  iconTestID,
}: ProfileAvatarProps) => {
  const pictureStyle = {
    width: 42,
    height: 42,
    borderRadius: 42,
    backgroundColor: '#2A2D3C',
  };

  const iconStyle = {
    position: 'absolute',
    left: 32,
    top: 22,
  } as StyleProp<ViewStyle>;

  return (
    <View style={styles.profileAvatar}>
      <Picture
        pictureFile={pictureFile}
        style={pictureStyle}
        testID={pictureTestID}
      />
      <Icon
        type={Icons.ELIPSE_PROFILE}
        width={20}
        height={20}
        style={iconStyle}
        testID={iconTestID}
      />
    </View>
  );
};
