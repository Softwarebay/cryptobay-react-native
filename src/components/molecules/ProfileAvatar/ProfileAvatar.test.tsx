import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';

import { ProfileAvatar } from './ProfileAvatar';

describe('ProfileAvatar Molecule', () => {
  test('Component Snapshot', () => {
    const tree = renderer.create(<ProfileAvatar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure custom styles works correctly', () => {
    render(
      <ProfileAvatar
        pictureTestID="profile-avatar-picture"
        iconTestID="profile-avatar-icon"
      />,
    );
    expect(screen.getByTestId('profile-avatar-picture')).toHaveStyle({
      width: 42,
      height: 42,
      borderRadius: 42,
      backgroundColor: '#2A2D3C',
    });
    expect(screen.getByTestId('profile-avatar-icon')).toHaveStyle({
      position: 'absolute',
      left: 32,
      top: 22,
    });
  });
});
