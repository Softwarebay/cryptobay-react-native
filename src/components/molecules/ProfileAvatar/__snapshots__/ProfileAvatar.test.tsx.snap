// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileAvatar Molecule Component Snapshot 1`] = `
<View
  style={
    Object {
      "position": "relative",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/pictures/profile.png",
      }
    }
    style={
      Object {
        "backgroundColor": "#2A2D3C",
        "borderRadius": 42,
        "height": 42,
        "width": 42,
      }
    }
  />
  <View
    style={
      Object {
        "left": 32,
        "position": "absolute",
        "top": 22,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="20"
      bbWidth="20"
      fill="none"
      focusable={false}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      testID="svg-icon-elipse-profile"
      vbHeight={16}
      vbWidth={16}
      width={20}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGCircle
          cx="8"
          cy="8"
          fill={
            Object {
              "payload": 4294885170,
              "type": 0,
            }
          }
          propList={
            Array [
              "fill",
            ]
          }
          r="8"
        />
        <RNSVGGroup>
          <RNSVGPath
            d="M6.5 10L7.5 11L6.5 12"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={
              Object {
                "payload": 4279703322,
                "type": 0,
              }
            }
          />
          <RNSVGPath
            d="M4 8.5V10C4 10.2652 4.10536 10.5196 4.29289 10.7071C4.48043 10.8946 4.73478 11 5 11H7.5"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={
              Object {
                "payload": 4279703322,
                "type": 0,
              }
            }
          />
          <RNSVGPath
            d="M9.5 6L8.5 5L9.5 4"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={
              Object {
                "payload": 4279703322,
                "type": 0,
              }
            }
          />
          <RNSVGPath
            d="M12 7.5V6C12 5.73478 11.8946 5.48043 11.7071 5.29289C11.5196 5.10536 11.2652 5 11 5H8.5"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={
              Object {
                "payload": 4279703322,
                "type": 0,
              }
            }
          />
          <RNSVGPath
            d="M6 7H4.5C4.23478 7 3.98043 6.89464 3.79289 6.70711C3.60536 6.51957 3.5 6.26522 3.5 6V4.5C3.5 4.23478 3.60536 3.98043 3.79289 3.79289C3.98043 3.60536 4.23478 3.5 4.5 3.5H6C6.26522 3.5 6.51957 3.60536 6.70711 3.79289C6.89464 3.98043 7 4.23478 7 4.5V6C7 6.26522 6.89464 6.51957 6.70711 6.70711C6.51957 6.89464 6.26522 7 6 7Z"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={
              Object {
                "payload": 4279703322,
                "type": 0,
              }
            }
          />
          <RNSVGPath
            d="M11.5 12.5H10C9.73478 12.5 9.48043 12.3946 9.29289 12.2071C9.10536 12.0196 9 11.7652 9 11.5V10C9 9.73478 9.10536 9.48043 9.29289 9.29289C9.48043 9.10536 9.73478 9 10 9H11.5C11.7652 9 12.0196 9.10536 12.2071 9.29289C12.3946 9.48043 12.5 9.73478 12.5 10V11.5C12.5 11.7652 12.3946 12.0196 12.2071 12.2071C12.0196 12.3946 11.7652 12.5 11.5 12.5Z"
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={
              Object {
                "payload": 4279703322,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
        <RNSVGDefs>
          <RNSVGClipPath
            name="clip0_4082_2735"
          >
            <RNSVGRect
              fill={
                Object {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              height="12"
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  2,
                  2,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
              width="12"
              x="0"
              y="0"
            />
          </RNSVGClipPath>
        </RNSVGDefs>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
